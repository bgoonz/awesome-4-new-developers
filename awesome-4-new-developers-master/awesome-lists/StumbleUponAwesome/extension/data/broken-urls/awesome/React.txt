http://blog.lum.pe/gesture-detection-in-react-native-fixing-unexpected-panning/,Gesture detection in React Native â€“ fixing unexpected panning,https://github.com/enaqx/awesome-react#readme,React error: URLError(gaierror(8, 'nodename nor servname provided, or not known')) | reason: [Errno 8] nodename nor servname provided, or not known
http://atticuswhite.com/blog/react-native-smart-mirror-lab/,Digital Smart Mirror lab with React Native,https://github.com/enaqx/awesome-react#readme,React error: timeout('The read operation timed out') | reason: 