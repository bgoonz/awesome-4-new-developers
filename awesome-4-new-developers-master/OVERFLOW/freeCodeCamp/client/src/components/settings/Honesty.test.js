import { Button } from '@freecodecamp/react-bootstrap';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import Honesty from './honesty';

describe('<Honesty />', () => {
  const renderer = new ShallowRenderer();
  const updateIsHonestMock = jest.fn();

  test('<Honesty /> snapshot when isHonest is false', () => {
    const componentToRender = (
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    );
    const view = renderer.render(componentToRender);
    expect(view).toMatchSnapshot('Honesty');
  });

  test('<Honesty /> snapshot when isHonest is true', () => {
    const componentToRender = (
      <Honesty isHonest={true} updateIsHonest={updateIsHonestMock} />
    );
    const view = renderer.render(componentToRender);
    expect(view).toMatchSnapshot('HonestyAccepted');
  });

  test('should call updateIsHonest method on clicking agree button', () => {
    const root = TestRenderer.create(
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    ).root;

    /**
     * This rules had to be disabled because the new lint rules are throwing false positives here.
     * They were interpreting react-test-renderer functions as @testing-library/react functions.
     */
    // eslint-disable-next-line testing-library/await-async-query
    root.findByType(Button).props.onClick();
    expect(updateIsHonestMock).toHaveBeenCalledWith({ isHonest: true });
  });
});
