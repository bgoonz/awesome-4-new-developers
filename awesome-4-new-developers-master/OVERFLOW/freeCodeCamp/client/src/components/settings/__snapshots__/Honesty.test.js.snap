// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Honesty /> <Honesty /> snapshot when isHonest is false: Honesty 1`] = `
<section
  className="honesty-policy"
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <HonestyPolicy />
    </Uncontrolled(Panel)>
    <br />
    <Button
      active={false}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      buttons.agree
    </Button>
  </FullWidthRow>
</section>
`;

exports[`<Honesty /> <Honesty /> snapshot when isHonest is true: HonestyAccepted 1`] = `
<section
  className="honesty-policy"
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <HonestyPolicy />
    </Uncontrolled(Panel)>
    <br />
    <Button
      active={false}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      className="disabled-agreed"
      disabled={true}
    >
      <p>
        buttons.accepted-honesty
      </p>
    </Button>
  </FullWidthRow>
</section>
`;
